commit 58294acf535a2378e5037974e8420cb873be0cb6
Author: David Faure <faure@kde.org>
Date:   Fri Jan 10 14:26:11 2014 +0100

    Add trailing slash when deleting a webdav folder
    
    As discussed on kde-devel, originally from https://forum.kde.org/viewtopic.php?f=59&t=119075

diff --git a/kioslave/http/http.cpp b/kioslave/http/http.cpp
index 42ccc94..b13eed1 100644
--- a/kioslave/http/http.cpp
+++ b/kioslave/http/http.cpp
@@ -1443,8 +1443,12 @@ void HTTPProtocol::rename( const KUrl& src, const KUrl& dest, KIO::JobFlags flag
     davError();
 }
 
-void HTTPProtocol::del( const KUrl& url, bool )
+void HTTPProtocol::del(const KUrl& _url, bool isFile)
 {
+  KUrl url(_url);
+  if (!isFile) {
+    url.adjustPath(KUrl::AddTrailingSlash);
+  }
   kDebug(7113) << url;
 
   if (!maybeSetRequestUrl(url))
